src/components/decimal/decimal.pw.tsx

```diff
@@ -0,0 +1,630 @@
+    mount,
+    page,
+  }) => {
+    await mount(<Decimal />);
+
+    const commonDataElementInputPreviewElement = commonDataElementInputPreview(
+      page
+    );
+    await commonDataElementInputPreviewElement.type("   ");
+    await commonDataElementInputPreviewElement.blur();
+    await expect(commonDataElementInputPreviewElement).toHaveAttribute(
+      "value",
+      "   "
+    );
+  });
+});
+
+test.describe("allowEmptyValue", () => {
+  ([
+    [0, "en", "0"],
+    [1, "en", "0.0"],
+    [2, "en", "0.00"],
+    [0, "es-ES", "0"],
+    [1, "es-ES", "0,0"],
+    [2, "es-ES", "0,00"],
+    [0, "fr", "0"],
+    [1, "fr", "0,0"],
+    [2, "fr", "0,00"],
+    [0, "pt-PT", "0"],
+    [1, "pt-PT", "0,0"],
+    [2, "pt-PT", "0,00"],
+  ] as [DecimalProps["precision"], string, string][]).forEach(
+    ([precisionValue, localeValue, expectedValue]) => {
+      test(`should format an empty value correctly when precision is ${precisionValue}, locale is ${localeValue} and allowEmptyValue is false`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(
+          <Decimal
+            precision={precisionValue}
+            locale={localeValue}
+            allowEmptyValue={false}
+          />
+        );
+
+        const commonDataElementInputPreviewElement = commonDataElementInputPreview(
+          page
+        );
+        await commonDataElementInputPreviewElement.fill("100");
+        await commonDataElementInputPreviewElement.clear();
+        await commonDataElementInputPreviewElement.blur();
+        await expect(commonDataElementInputPreviewElement).toHaveAttribute(
+          "value",
+          expectedValue
+        );
+      });
+    }
+  );
+
+  ([
+    [0, "en"],
+    [1, "en"],
+    [2, "en"],
+    [0, "es-ES"],
+    [1, "es-ES"],
+    [2, "es-ES"],
+    [0, "fr"],
+    [1, "fr"],
+    [2, "fr"],
+    [0, "pt-PT"],
+    [1, "pt-PT"],
+    [2, "pt-PT"],
+  ] as [DecimalProps["precision"], string][]).forEach(
+    ([precisionValue, localeValue]) => {
+      test(`should format an empty value correctly when precision is ${precisionValue}, locale is ${localeValue} and allowEmptyValue is true`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(
+          <Decimal
+            precision={precisionValue}
+            locale={localeValue}
+            allowEmptyValue
+          />
+        );
+
+        const commonDataElementInputPreviewElement = commonDataElementInputPreview(
+          page
+        );
+        await commonDataElementInputPreviewElement.fill("100");
+        await commonDataElementInputPreviewElement.clear();
+        await commonDataElementInputPreviewElement.blur();
+
+        await expect(commonDataElementInputPreviewElement).toHaveAttribute(
+          "value",
+          ""
+        );
+      });
+    }
+  );
+});
+
+test.describe("check events for Decimal component", () => {
+  const inputValue = "123";
```

Comment:
**suggestion(non-blocking):** As `inputValue` is only used in the `should call onBlur callback when a blur event is triggered` test, you could move this from here to inside of that test block. Having it here does no harm, so if the QA reviewer wants you to keep it here as you've agreed this pattern of declaring `const`'s, I'm fine with that.

Reply:
Well spotted, I've moved that const within that test case now.